File: code/video/MyCloudMusicAndroidJava/app/src/main/java/com/ixuea/courses/mymusic/util/MessageUtil.java
package com.ixuea.courses.mymusic.util;

import org.apache.commons.lang3.StringUtils;

import io.rong.imlib.model.MessageContent;
import io.rong.message.ImageMessage;
import io.rong.message.TextMessage;

/**
 * 消息工具类
 */
public class MessageUtil {
    public static String getContent(MessageContent messageContent) {
        if (messageContent instanceof TextMessage) {
            return ((TextMessage) messageContent).getContent();
        } else if (messageContent instanceof ImageMessage) {
            return "[图片]";
        }

        return "";
    }

    public static String getNickname(String id, String nickname) {
        return StringUtils.isNotBlank(nickname) ? nickname : id;
    }

    public static String createPushData(String data, String userId) {
        return "";
    }
}

File: code/video/MyCloudMusicAndroidJava/app/src/main/java/com/ixuea/courses/mymusic/util/MessageUtil.kt
package com.ixuea.courses.mymusic.util

import io.rong.imlib.model.MessageContent
import io.rong.message.ImageMessage
import io.rong.message.TextMessage
import org.apache.commons.lang3.StringUtils

/**
 * 消息工具类
 */
object MessageUtil {
    @JvmStatic
    fun getContent(messageContent: MessageContent?): String {
        return when (messageContent) {
            is TextMessage -> messageContent.content
            is ImageMessage -> "[图片]"
            else -> ""
        }
    }

    @JvmStatic
    fun getNickname(id: String, nickname: String?): String {
        return if (StringUtils.isNotBlank(nickname)) nickname.orEmpty() else id
    }

    @JvmStatic
    @Suppress("UNUSED_PARAMETER")
    fun createPushData(_data: String, _userId: String): String {
        return ""
    }
}
